src/evaluator/functions.rs

```diff
@@ -1658,3 +1658,60 @@ fn multiply_by_pow10(num: f64, pow: isize) -> Result<f64> {
         .parse::<f64>()
         .map_err(|e| Error::D3137Error(e.to_string()))
 }
+
+pub fn fn_pad<'a>(
+    context: FunctionContext<'a, '_>,
+    args: &[&'a Value<'a>],
+) -> Result<&'a Value<'a>> {
+    let str_value = args.first().copied().unwrap_or_else(Value::undefined);
+    if str_value.is_undefined() {
+        return Ok(Value::undefined());
+    }
+
+    let str_to_pad = if str_value.is_string() {
+        str_value.as_str() // as_str returns Cow<'_, str>
+    } else {
+        return Ok(Value::undefined());
+    };
+
+    let width_value = args.get(1).copied().unwrap_or_else(Value::undefined);
+    if !width_value.is_number() {
+        return Ok(Value::undefined());
+    }
+
+    let width_i64 = width_value.as_f64().round() as i64;
+
+    // Extract or default to space, treat empty string as space
+    let pad_char = args
+        .get(2)
+        .map(|v| v.as_str())
+        .filter(|c| !c.is_empty())
+        .unwrap_or(Cow::Borrowed(" "));
+
+    let pad_length = (width_i64.unsigned_abs() as usize).saturating_sub(str_to_pad.chars().count());
+
+    // Early return if no padding is needed
+    if pad_length == 0 {
+        return Ok(Value::string(context.arena, &str_to_pad));
+    }
+
+    // Create padding string, handle multi-character pad strings with truncation
+    let padding = if pad_char.chars().count() > 1 {
+        pad_char
+            .chars()
+            .cycle()
+            .take(pad_length)
+            .collect::<String>()
+    } else {
+        pad_char.repeat(pad_length)
+    };
+
+    // Depending on width, pad left or right
+    let result = if width_i64 > 0 {
+        format!("{}{}", str_to_pad, padding)
```

Comment:
I think we can use the standard library to pad for us.

https://doc.rust-lang.org/std/fmt/index.html#fillalignment

Reply:
yes, padding is handled by rust lib's supported fmt.